### PR TITLE
ScaleGesture + added a scale animator

### PR DESCRIPTION
**Pull request by saniul**
This introduces a `ScaleGesture` and adds a `scale` animator to `Layer`

### CODE REVIEW

**Comment by saniul** [comment]
So this is ready to merge too, I guess :)

File: Prototope/Gesture.swift [hunk 1]
```diff
@@ -294,6 +294,93 @@ public class RotationGesture: GestureType {
...
 }
 
+/** A pinch sample represents the state of a pinch gesture at a single point in time */
+public struct PinchSample: SampleType {
+    public let scale: Double
+    public let velocity: Double
+    public let centroid: TouchSample
+    
+    public var description: String {
+        return "{PinchSample: scale:\(scale) velocity\(velocity) @\(centroid)}"
```
Comment (andymatuschak) [inline Prototope/Gesture.swift]: Little nit: could you use `scale: \(scale) velocity: \(velocity)`?

File: Prototope/Animation.swift [hunk 2]
```diff
@@ -48,6 +48,8 @@ public class LayerAnimatorStore {
 	public var backgroundColor: Animator<Color>
 	public var alpha: Animator<Double>
 	public var rotationRadians: Animator<Double>
+    //kPOPScaleXY expects two values, not one!
+    public var scale: Animator<Point>
```
Comment (andymatuschak) [inline Prototope/Animation.swift]: Looks like there are some whitespace hiccups in this file.

**Comment by andymatuschak** [comment]
Looking really good!

File: Prototope/Animation.swift (hunk 2 above)
Comment (saniul) [inline Prototope/Animation.swift]: Hm, that’s GIthub’s rendering messing it up - I use spaces and you used tabs, it seems :)

**Comment by saniul** [comment]
Done.

**Comment by saniul** [comment]
Just a heads up, I opened an [Issue for simultaneous recognition of Gestures](https://github.com/Khan/Prototope/issues/6). I’d like to have a single scene that combines both Rotation and Scale, but wanted to gather some feedback on the architecture first.

**Comment by andymatuschak** [comment]
Looks great, thanks!

**Pull request merged**